addons/mail/models/mail_thread.py

```diff
@@ -1110,21 +1119,27 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         msg_references = [ref for ref in tools.mail_header_msgid_re.findall(thread_references) if 'reply_to' not in ref]
         mail_messages = MailMessage.sudo().search([('message_id', 'in', msg_references)], limit=1)
         is_a_reply = bool(mail_messages)
+        alias_domain = [('alias_name', 'in', rcpt_tos_localparts)]
 
         # 1.1 Handle forward to an alias with a different model: do not consider it as a reply
         if reply_model and reply_thread_id:
-            other_alias = Alias.search([
+            other_aliases = Alias.search([
                 '&',
                 ('alias_name', '!=', False),
-                ('alias_name', '=', email_to_localpart)
+                ('alias_name', 'in', email_to_localparts),
             ])
-            if other_alias and other_alias.alias_model_id.model != reply_model:
-                is_a_reply = False
+            for other_alias in other_aliases:
```

Comment:
Considering the other comments, maybe we could get simplify a bit code here. I think that

  * either there is at least one alias of the same model -> is_a_reply takes the hand
    * is_a_reply is True: filter aliases only of current model
    * is_a_reply is False: actually I am not sure this should really happen, but anyway we will apply aliases
  * either all aliases are linked to another model -> we will apply aliases

In which case it could be: if all (other_alias.model_id.model != reply_model for other_alias in other_aliases): is_a_reply = False

As it is a bit tricky, could you confirm :p ?

Reply:
I actually managed to simplify it even more, as you can see in the last version. Basically just ask the DB: is there an alias with this name and this model? Then it's a reply.